src/components/search/list/TextList.tsx

```diff
@@ -1,99 +1,120 @@
-import { forwardRef } from 'react';
+import { useEffect, useState } from 'react';
 import Image from 'next/image';
 import Link from 'next/link';
+import useLocalStorage from 'use-local-storage';
 
 import Loading from '@/app/loading';
-import DefaultImg from '@/components/img/DefaultImg';
+import DefaultImg from '@/components/common/empty/DefaultImg';
 import StarImg from '@/components/svg/ico_star.svg';
 import { ListData } from '@/types/search.type';
 
 interface TextListProps {
   tab: string;
   list: ListData[];
-  isLoading: boolean;
+  filters?: string[];
 }
 
-const TextList = forwardRef<HTMLDivElement, TextListProps>(
-  ({ tab, list }, ref) => {
-    if (!Array.isArray(list)) return <Loading />;
-    return (
-      <ul>
-        {list.map((item) => {
-          return (
-            <li
-              key={item.contentId}
-              className="flex items-center border-solid border-b-[1px] border-gray-300"
+const FoodFilter = {
+  전체: 'all',
+  한식: 'korean',
+  양식: 'western',
+  일식: 'japanese',
+  중식: 'chinese',
+  카페: 'cafe',
+} as const;
+const TextList = ({ tab, list, filters }: TextListProps) => {
+  const [filterList, setFilterList] = useState(list);
+  const [_, setScrollY] = useLocalStorage('places_list_scroll', 0);
+
+  useEffect(() => {
```

Comment:
`useEffect` 는 list와 filters가 변경 될 때 호출 되는것으로 보여요.
이는 사실상 부모로부터 받아온 Props가 변경 되면 filterList를 다시 재계산한다고도 볼 수 있을 것 같아요.
그렇다면 useEffect를 사용하는 것보다는 일반 변수로 직접 계산 하는 것도 괜찮아보여요.

Reply:
useEffect를 사용하지 않았더니 Error: Too many re-renders. React limits the number of renders to prevent an infinite loop. 이런 에러가 떠서 제가 useEffect로 list랑 filters를 사용했는데요 이 에러에 대해서 useEffect가 근본적인 문제 해결방법이 아닌걸까요~?

Reply:
list도 useState를 제거하니까 연쇄적으로 useEffect를 사용하지 않게 되었어요! 감사합니다~